### PR TITLE
Fixes impossible radio uplink codewords for lizardpeople

### LINKED ISSUE #9932: [Bug]: Lizards can't correctly speak radio uplink codewords with an S in them
### BYOND Version

_No response_

### Round Date

2023/09/29

### Round ID

46049

### Test merges

- #9124

### Describe the bug

1. As a lizard traitor, I had a radio uplink with the codeword "Oscar". I couldn't open the uplink because my lizard's accent made it come out as "Ossscar." I was forced to get a fellow traitor to open my uplink for me.

### What did you expect?

I expect the uplink to work despite a lizard's accent.

### Reproducible steps

1. Be a lizard traitor.
2. Try to use a radio uplink with an S present in the codeword.

### PR DESCRIPTION
**Pull request by BeeLover66**
## About The Pull Request

This PR closes https://github.com/BeeStation/BeeStation-Hornet/issues/9932

A lizardperson couldn't unlock a radio uplink when the codeword was Oscar, Whiskey or X-ray because of their accent, as the uplink wouldn't recognize "Ossscar", "Whissskey" and "ECKS-ray" as valid. This PR removes the accent before parsing the codeword.

## Why It's Good For The Game

This is a bug, lizardpeople should be able to unlock their uplink.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Oscar
oscar
Whiskey
whiskey
X-ray
x-ray

</details>

## Changelog
:cl:
fix: Lizardperson's accent no longer prevents them from unlocking a radio uplink when the codeword contains an s or an x
/:cl:

### CODE REVIEW

**Review by itsmeow — APPROVED** [review]
Unfortunately, as much as this solution is janky, having looked into this code extensively, it is not really possible to get the "untreated message" to this signal without excessive parameter passing. This is the best solution available. I do hate it though.

**Comment by CydiaLamiales** [comment]
> This is the best solution available. I do hate it though.

What about modifying the code message based on the user's tongue speech modifiers when the uplink is created?

**Comment by BeeLover66** [comment]
> What about modifying the code message based on the user's tongue speech modifiers when the uplink is created?
That would be possible, but the problem would remain in cases where a lizardperson tries to unlock a non-lizard's uplink or vice versa.

**Comment by CydiaLamiales** [comment]
> > What about modifying the code message based on the user's tongue speech modifiers when the uplink is created?
> 
> That would be possible, but the problem would remain in cases where a lizardperson tries to unlock a non-lizard's uplink or vice versa.

Non-lizardpeople could mimic the accent by manually changing the letters. As for the other way, the lizardperson could get their tongue swapped or use someone else to unlock it.
Afterall, it's not their uplink.

**Comment by BeeLover66** [comment]
> Non-lizardpeople could mimic the accent by manually changing the letters. As for the other way, the lizardperson could get their tongue swapped or use someone else to unlock it. Afterall, it's not their uplink.

I suppose that makes sense. I've made some changes locally to do it that way, but I don't know wether I should push them considering the PR is already approved.

This is what the new version would look like : 
ECKS-ray codeword

**Comment by itsmeow** [comment]
Funny, but the syndicate is definitely specist and would never do such a thing

**Comment by BeeLover66** [comment]
> Funny, but the syndicate is definitely specist and would never do such a thing

On the other hand, you could argue that it'd be in character for the syndicate to give lizards accented codewords instead of using an inclusive speech recognition system...

I'll keep the old solution though

**Comment by Howluinb** [comment]
Can't you just take the janky af way and add a "sss" and "ks" of every word in the codeword list?

**Comment by BeeLover66** [comment]
> Can't you just take the janky af way and add an extra "sss" and "ks" version of every word in the codeword list?

I'm not sure what you're referring to, can you rephrase your suggestion?

**Comment by Howluinb** [comment]
Like, if a codeword is stated to be "Oscar", make the code ALSO recognise Ossscar as a valid input too, and this accross all codewords

**Comment by BeeLover66** [comment]
> Like, if a codeword is stated to be "Oscar", make the code ALSO recognise Ossscar as a valid input too, and this accross all codewords

That's probably technically possible, but I feel like the implementation would be as janky as what I have now, if not more

**Review by EvilDragonfiend — CHANGES_REQUESTED** [review]
since you're here, please add flyperson zzz ones too.

**Comment by BeeLover66** [comment]
> since you're here, please add flyperson zzz ones too.

Done, as well as "kss" for lizardpeople. Those two would only be used if an admin sets a custom codeword though, assuming we keep the phonetic alphabet as default.

**Comment by EvilDragonfiend** [comment]
I wasn't confident to explain how you can fix without using regex, so I posted my version to fix.

**Comment by BeeLover66** [comment]
> I wasn't confident to explain how you can fix without using regex, so I posted my version to fix.

Huh, so there was a non-janky way to do it after all. Thanks a lot!
